code/io/keycontrol.cpp

```diff
@@ -1888,8 +1888,9 @@ int button_function_critical(int n, net_player *p = NULL)
 
 			int firepoints = pm->missile_banks[Ships[objp->instance].weapons.current_secondary_bank].num_slots;
 
-            if (Ships[objp->instance].flags[Ship::Ship_Flags::Secondary_dual_fire] || firepoints < 2) {
-                Ships[objp->instance].flags.remove(Ship::Ship_Flags::Secondary_dual_fire);
+			if (Ships[objp->instance].flags[Ship::Ship_Flags::Secondary_dual_fire] || firepoints < 2 ||
+				The_mission.ai_profile->flags[AI::Profile_Flags::Disable_player_secondary_doublefire]) {
+				Ships[objp->instance].flags.remove(Ship::Ship_Flags::Secondary_dual_fire);
```

Comment:
To the first, my thinking was that it might be set if the player previously had it set on a save where this flag wasn't used. I'm frankly not certain if that's possible, but it seemed harmless to be sure. Strictly speaking this is safeguarded by the check at firing time, but doing it here provides more immediate player feedback on the possibility of firing two missiles. 

To the second, I don't understand how that could be the case. I'm also not sure it's especially undesirable, again giving the player more immediate information.

On the gripping hand though this doesn't currently handle the case of the individual weapon flag, so even if I prevail on the first two points it still needs work, though different work.

Reply:
Again, use an Assertion.

"every time the mission began" is not strictly true; let me rephrase.  This change would take a block of code that's not supposed to run and not expected to run, due to the lack of secondary linking.  But when you check the ai_profiles flag it *does* run and then immediately reverts what it did.  This is bad logic.